traits/trait_types.py

```diff
@@ -1516,6 +1528,10 @@ def create_editor(self):
 class Directory(BaseDirectory):
     """ A fast-validating trait type whose value is a directory path string.
 
+    For Python 3.6 and greater, it also accepts objects implementing
+    the :class: `os.PathLike` interface, converting them to the corresponding
```

Comment:
```suggestion
    the :class:`os.PathLike` interface, converting them to the corresponding
```

Reply:
committed the suggestions.

Reply:
Thank you! Please merge, if you're happy with the current state of the PR.